service/history/workflowTaskHandler.go

```diff
@@ -199,12 +200,19 @@ func (handler *workflowTaskHandlerImpl) handleCommands(
 	return mutations, nil
 }
 
-func (handler *workflowTaskHandlerImpl) handleCommand(ctx context.Context, command *commandpb.Command) (*handleCommandResponse, error) {
+func (handler *workflowTaskHandlerImpl) handleCommand(
+	ctx context.Context,
+	command *commandpb.Command,
+	earlyDeliverMessages func(context.Context) error,
+) (*handleCommandResponse, error) {
 	switch command.GetCommandType() {
 	case enumspb.COMMAND_TYPE_SCHEDULE_ACTIVITY_TASK:
 		return handler.handleCommandScheduleActivity(ctx, command.GetScheduleActivityTaskCommandAttributes())
 
 	case enumspb.COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION:
+		if err := earlyDeliverMessages(ctx); err != nil {
+			return nil, err
+		}
```

Comment:
@alexshtin this what gets us around the problem discovered last night - early deliver messages in this one case so that update completion is visible to the command handler.